sklearn/svm/src/libsvm/svm.cpp

```diff
@@ -3143,12 +3143,15 @@ const char *PREFIX(check_parameter)(const PREFIX(problem) *prob, const svm_param
 		// filter samples with negative and null weights 
 		remove_zero_weight(&newprob, prob);
 
-		char* msg = NULL;
 		// all samples were removed
-		if(newprob.l == 0)
-			msg =  "Invalid input - all samples have zero or negative weights.";
+		if(newprob.l == 0) {
+			free(newprob.x);
+			free(newprob.y);
+			free(newprob.W);
+			return "Invalid input - all samples have zero or negative weights.";
+		}
 		else if(prob->l != newprob.l && 
-		        svm_type == C_SVC)
+				svm_type == C_SVC)
```

Comment:
For git blame purposes, can you revert this change? (The rest of this file lines up conditionals this way)

Reply:
fb942be5a2c8a47a35fe35891e25981a73b08a58 revert this change.

Reply:
Done via d58cc42cb69306ecb3c9f685f8b05c67d7047d54.